hr_holidays_natural_period/tests/test_hr_leave.py

```diff
@@ -0,0 +1,137 @@
+            }
+        )
+        cls.user = new_test_user(cls.env, login="test-user")
+        cls.employee = cls.env["hr.employee"].create(
+            {
+                "name": "Test employee",
+                "address_id": partner.id,
+                "resource_calendar_id": calendar.id,
+                "user_id": cls.user.id,
+            }
+        )
+
+    def _create_leave_allocation(self, leave_type, days):
+        leave_allocation_form = Form(
+            self.env["hr.leave.allocation"].with_context(
+                default_date_from="2023-01-01",
+                default_date_to=f"{fields.Date.today().year}-12-31",
+            )
+        )
+        leave_allocation_form.holiday_status_id = leave_type
+        leave_allocation_form.number_of_days_display = days
+        return leave_allocation_form.save()
+
+    def _create_hr_leave(self, leave_type, date_from, date_to):
+        leave_form = Form(self.env["hr.leave"])
+        leave_form.holiday_status_id = leave_type
+        leave_form.request_date_from = date_from
+        leave_form.request_date_to = date_to
+        return leave_form.save()
+
+    @users("test-user")
+    @freeze_time("2023-01-01", tick=True)
+    @mute_logger("odoo.models.unlink")
+    def test_hr_leave_natural_day_01(self):
+        leave_allocation = self._create_leave_allocation(self.leave_type, 5)
+        leave_allocation.sudo().action_validate()
+        res_leave_type = (
+            self.env["hr.leave.type"]
+            .with_company(self.env.company)
+            .get_allocation_data_request()[0][1]
+        )
+        self.assertEqual(res_leave_type["remaining_leaves"], 5)
+        self.assertEqual(res_leave_type["virtual_remaining_leaves"], 5)
+        self.assertEqual(res_leave_type["max_leaves"], 5)
+        self.assertEqual(res_leave_type["leaves_taken"], 0)
+        self.assertEqual(res_leave_type["virtual_leaves_taken"], 0)
+        self.assertEqual(res_leave_type["request_unit"], "natural_day")
+        leave = self._create_hr_leave(self.leave_type, "2023-01-02", "2023-01-05")
+        self.assertEqual(leave.number_of_days, 4.0)
+
+    @users("test-user")
+    @freeze_time("2023-01-01", tick=True)
+    @mute_logger("odoo.models.unlink")
+    def test_hr_leave_natural_day_02(self):
+        attendances = [(0, 16, 21), (1, 9, 14), (2, 9, 14), (3, 9, 14), (4, 9, 14)]
+        r_sudo = self.env["resource.calendar"].sudo()
+        calendar = r_sudo.create(
+            {
+                "name": "Test calendar",
+                "tz": "Europe/Brussels",
+                "attendance_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": index,
+                            "dayofweek": str(att[0]),
+                            "hour_from": att[1],
+                            "hour_to": att[2],
+                        },
+                    )
```

Comment:
Could you please use `Command`? It's more readable
https://github.com/odoo/odoo/blob/f45451b98e2e12aa803f1abc0ba3c9a6f90310f3/odoo/fields.py#L4227
```suggestion
                    Command.create(
                        {
                            "name": index,
                            "dayofweek": str(att[0]),
                            "hour_from": att[1],
                            "hour_to": att[2],
                        },
                    )
```

Reply:
Yes, of course. I have to get used to use it, it's more readable, for the moment, I'm old school!